core/tests/puppeteer-acceptance-tests/user-utilities/blog-post-admin-utils.js

```diff
@@ -0,0 +1,461 @@
+// Copyright 2023 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Blog Admin users utility file.
+ */
+
+const puppeteerUtilities = require('../puppeteer-testing-utilities/puppeteer-utils.js');
+const testConstants = require(
+  '../puppeteer-testing-utilities/test-constants.js');
+const { showMessage } = require('../puppeteer-testing-utilities/show-message-utils.js');
+
+const blogTitleInput = 'input.e2e-test-blog-post-title-field';
+const blogBodyInput = 'div.e2e-test-rte';
+const thumbnailPhotoBox = 'div.e2e-test-photo-clickable';
+const unauthErrorContainer = 'div.e2e-test-error-container';
+const blogDashboardAuthorDetailsModal = 'div.modal-dialog';
+const roleUpdateUsernameInput = 'input#label-target-update-form-name';
+const blogEditorUsernameInput = 'input#label-target-form-reviewer-username';
+const maximumTagLimitInput = 'input#mat-input-0';
+const blogAuthorBioField = 'textarea.e2e-test-blog-author-bio-field';
+const blogDashboardUrl = testConstants.URLs.BlogDashboard;
+const blogAdminUrl = testConstants.URLs.BlogAdmin;
+const LABEL_FOR_NEW_BLOG_POST_CREATE_BUTTON = ' CREATE NEW BLOG POST ';
+const LABEL_FOR_SAVE_BUTTON = ' SAVE ';
+const LABEL_FOR_DONE_BUTTON = ' DONE ';
+const LABEL_FOR_SAVE_DRAFT_BUTTON = 'SAVE AS DRAFT';
+const LABEL_FOR_DELETE_BUTTON = 'Delete';
+const LABEL_FOR_CONFIRM_BUTTON = ' Confirm ';
+const LABEL_FOR_ADD_THUMBNAIL_BUTTON = ' Add Thumbnail Image ';
+const LABEL_FOR_ADD_ELEMENT_BUTTON = ' Add element ';
+
+module.exports = class e2eBlogPostAdmin extends puppeteerUtilities {
+  /**
+   * Fucntion for adding blog post author bio in blog dashboard.
+   */
+  async addUserBioInBlogDashboard() {
+    await this.type(blogAuthorBioField, 'Dummy-User-Bio');
+    await this.page.waitForSelector(
+      'button.e2e-test-save-author-details-button:not([disabled])');
+    await this.clickOn(LABEL_FOR_SAVE_BUTTON);
+  }
+
+  /**
+   * Function for navigating to the blog dashboard page.
+   */
+  async navigateToBlogDashboardPage() {
+    await this.goto(blogDashboardUrl);
+  }
+
+  /**
+   * This function creates a blog post with given title.
+   * @param {string} draftBlogPostTitle - The title of the draft blog post
+   * to be created.
+   */
+  async createDraftBlogPostWithTitle(draftBlogPostTitle) {
+    await this.addUserBioInBlogDashboard();
+    // See Note-1 below.
```

Comment:
Don't do this. Have comments be next to where they're read.

Reply:
done